jerry-core/api/jerry.c

```diff
@@ -254,6 +255,44 @@ jerry_gc (void)
   ecma_gc_run (JMEM_FREE_UNUSED_MEMORY_SEVERITY_LOW);
 } /* jerry_gc */
 
+#ifdef JMEM_STATS
+void jmem_heap_get_stats (jmem_heap_stats_t *);
+#endif
+/**
+ * Get heap memory stats.
+ *
+ * @return true - get the heap stats successful
+ *         false - otherwise. Usually it is because the MEM_STATS feature is not enabled.
+ */
+bool
+jerry_get_memory_stats (jerry_heap_stats_t *out_stats_p) /**< [out] heap memory stats */
+{
+#ifdef JMEM_STATS
+  if (out_stats_p == NULL)
+  {
+    return false;
+  }
+
+  jmem_heap_stats_t jmem_heap_stats = {0};
+  jmem_heap_get_stats (&jmem_heap_stats);
+
```

Comment:
Thinking this further:

Say today `sizeof(jerry_heap_stats_t) == 24`.
Say I dynamically link my project against JerryScript of today, so my project does something like:
```
jerry_heap_stats_t stats = {}; // 24 bytes allocated on the stack
jerry_get_memory_stats(&stats); // 24 bytes written
```

Now tomorrow I have a newer JerryScript, which happened to have added some fields to `jerry_heap_stats_t`, so now `sizeof(jerry_heap_stats_t) == 36`.

Now it will be a problem if the project dynamically links to the newer JerryScript:
```
jerry_heap_stats_t stats = {}; // still 24 bytes allocated on the stack
jerry_get_memory_stats(&stats); // 36 bytes written -- WHOOPS, smashing the stack!
```

That's what I meant with "I think the caller should be passing in the desired/expected version then for this to work" in [my earlier comment].

cc @zherczeg 

[my earlier comment]: https://github.com/jerryscript-project/jerryscript/pull/1923#discussion_r128181423